as/cNFT/assembly/mint.ts

```diff
@@ -34,11 +34,14 @@ export function mint(tokenMetadata: TokenMetadata, token_royalty: TokenRoyalty):
     persistent_tokens_royalty.add(tokenId, token_royalty)
 
 
+    token.metadata = tokenMetadata
```

Comment:
This invalidates having a `PersistentTokenMetadata` if you are just gonna save it here. I would try to remove this.